File: [프로그래머스] 2019 KAKAO BLIND RECRUITMENT_오픈채팅방.js
function solution(record) {
    var answer = [];
    let userId = {};
    let status = {"Enter":"님이 들어왔습니다.", "Leave":"님이 나갔습니다."};
    for(let i of record){
        const arr = i.split(' ');
        if(arr[0] === "Enter" || arr[0] === "Change"){
            userId[arr[1]] = arr[2];
        }
    }
    for(let i of record){
        const arr = i.split(' ');
        if(arr[0] !== "Change"){
            answer.push(userId[arr[1]]+status[arr[0]]);
        }
    }
    return answer;
}

File: [프로그래머스] 2019 KAKAO BLIND RECRUITMENT_오픈채팅방.py
def solution(record):
    answer = []
    dic = {}
    arr = []
    for i in record:
        arr = i.split()
        if arr[0] == "Enter" or arr[0] == "Change":
            dic[arr[1]] = arr[2]
    for i in record:
        arr = i.split()
        if arr[0] == "Enter":
            answer.append(dic[arr[1]]+"님이 들어왔습니다.")
        elif arr[0] == "Leave":
            answer.append(dic[arr[1]]+"님이 나갔습니다.")
    return answer